internal/agent/device/lifecycle/manager.go

```diff
@@ -97,6 +98,14 @@ func (m *LifecycleManager) Initialize(ctx context.Context, status *v1alpha1.Devi
 		}
 	}
 
+	_, updateErr := m.statusManager.Update(ctx, fcstatus.SetDeviceLifecycleStatus(v1alpha1.DeviceLifecycleStatus{
+		Status: "Enrolled",
```

Comment:
This should be a const to eliminate future typos but I am trying to understand why this is necessary. The service knows if the agent is enrolled already and the agent could lie if it wanted to.

Reply:
The main question is who owns this status? Should it be the agent or the service.

Reply:
This is a good point re: owning the status. We did decide the service would own it. However, it is getting wiped for some reason even after being set on the service side - I'm not completely sure but I think this may be because a blank status is created for the device during bootstrap if one does not exist already. Maybe a better solution would be to make sure the service-side status gets passed to the agent upon enrollment (so there is never a case where a device status doesn't exist)? I'm open to suggestions.

The related bug is [this one](https://redhat-internal.slack.com/archives/C01KER8QV33/p1738313965026339). This change did fix the issue for me when I tested it.

Reply:
https://github.com/flightctl/flightctl/blob/main/api/v1alpha1/constructors.go#L29-L31 this should be removed so the status is not sent from the agent at all.

Reply:
That code is being used on the server-side, though.

Reply:
This is where the device status gets set initially on the service side: https://github.com/flightctl/flightctl/blob/main/internal/service/enrollmentrequest.go#L81 . But I do not know whether that initial "device status" is ever sent to the agent - there isn't a reason to do it as the status is normally sent from agent to service. But in this case I think the "new" device status is spawned again on the device side and overwrites the previous status.

Reply:
it's also being used by the agent in this case I would unset it after here[1] then.

[1] https://github.com/flightctl/flightctl/blob/dfc4b3ba042b21b5824082f29803dc77028f8a62/internal/agent/device/status/status.go#L22